### PR TITLE
Fix AGFS_COMMITTAL bill types and conditionally pass "Advocate Fee"

### PR DESCRIPTION
**Pull request by jsugarman**
**What**
- [X] Fix AGFS_COMMITTAL bill sub types
- [X] Only send advocate fee bill for applicable case types

**Why**
* typo in mappings for sub types
* certain claims types cannot have an Advocate Fee bill

### CODE REVIEW

File: spec/api/v2/ccr_claim_spec.rb [hunk 1]
```diff
@@ -107,14 +107,20 @@ def do_request(claim_uuid: @claim.uuid, api_key: @case_worker.user.api_key)
...
+      end
 
-      context 'bill type' do
-        subject(:response) do
-          do_request(claim_uuid: claim.uuid, api_key: @case_worker.user.api_key).body
-        end
+      let(:claim) { create(:authorised_claim) }
 
-        let(:claim) { create(:authorised_claim) }
+      context 'excludes advocate fees for certain case' do
```
Comment (jsugarman) [inline spec/api/v2/ccr_claim_spec.rb]: cases

File: spec/api/v2/ccr_claim_spec.rb [hunk 2]
```diff
@@ -107,14 +107,20 @@ def do_request(claim_uuid: @claim.uuid, api_key: @case_worker.user.api_key)
...
+        it 'returns nil for case types that do not permit advocate fees' do
+          allow_any_instance_of(CaseType).to receive(:fee_type_code).and_return 'FXCON' # mock a contempt case type
+          expect(response).not_to have_json_path("bills/0/bill_type")
```
Comment (jsugarman) [inline spec/api/v2/ccr_claim_spec.rb]: better way of checking number of entries in array?!

File: app/services/ccr/fee_adapter.rb [hunk 3]
```diff
@@ -34,14 +34,14 @@ def self.zip(bill_types = [])
...
+      FXCON: zip([nil, nil]), # Contempt
       GRRAK: zip(%w[AGFS_FEE AGFS_FEE]), # Cracked Trial
       GRCBR: zip(%w[AGFS_FEE AGFS_FEE]), # Cracked before retrial
-      GRDIS: zip(%w[NOT_ALLOWED NOT_ALLOWED]), # Discontinuance
-      FXENP: zip(%w[NOT_ALLOWED NOT_ALLOWED]), # Elected cases not proceeded
+      GRDIS: zip([nil, nil]), # Discontinuance
+      FXENP: zip([nil, nil]), # Elected cases not proceeded
       GRGLT: zip(%w[AGFS_FEE AGFS_FEE]), # Guilty plea
-      FXH2S: zip(%w[NOT_APPLICABLE NOT_APPLICABLE NOT_APPLICABLE]), # Hearing subsequent to sentence??? LGFS only
+      FXH2S: zip([nil, nil]), # Hearing subsequent to sentence??? LGFS only
```
Comment (colinbruce) [inline app/services/ccr/fee_adapter.rb:44]: There were three `NOT_APPLICABLE`s over there <= should there be three here?

**Review by colinbruce — APPROVED** [review]
One observation, but otherwise 👍

File: app/services/ccr/fee_adapter.rb (hunk 3 above)
Comment (jsugarman) [reply app/services/ccr/fee_adapter.rb:44]: the three was wrong - a left over from some spiked stuff

**Pull request merged**